csrc/abstract_tensor_schedule.cpp

```diff
@@ -0,0 +1,319 @@
+              concrete_ids_.emplace(vg, id_inps.front());
+            }
+            continue;
+          }
+          // Compute new ID expression
+          Expr* id_expr = nullptr;
+          if (auto* m = dynamic_cast<Merge*>(e)) {
+            NVF_ERROR(id_inps.size() == 2);
+            auto* new_id = IterDomain::merge(id_inps[0], id_inps[1]);
+            id_expr = new_id->definition();
+          } else if (auto* s = dynamic_cast<Split*>(e)) {
+            NVF_ERROR(id_inps.size() == 1);
+            auto new_ids =
+                IterDomain::split(id_inps[0], s->factor(), s->innerSplit());
+            id_expr = new_ids.first->definition();
+          } else if (auto* s = dynamic_cast<Swizzle*>(e)) {
+            NVF_ERROR(id_inps.size() == 2);
+            auto new_ids =
+                IterDomain::swizzle(s->swizzleType(), id_inps[0], id_inps[1]);
+            id_expr = new_ids.first->definition();
+          } else if (auto* s = dynamic_cast<Swizzle2D*>(e)) {
+            NVF_ERROR(id_inps.size() == 2);
+            auto new_ids = IterDomain::swizzle(
+                s->swizzleType(), id_inps[0], id_inps[1], s->swizzleMode());
+            id_expr = new_ids.first->definition();
+          } else {
+            NVF_ERROR(false, "Unhandled IterDomain expression ", e->toString());
+          }
+          // Update the mapping to point to all of the newly created Expr's
+          // outputs
+          NVF_ERROR(id_expr != nullptr);
+          NVF_ERROR(id_expr->outputs().size() == e->outputs().size());
+          for (size_t i : c10::irange(e->outputs().size())) {
+            ValGroup vg_outp = graph_->toGroup(e->output((int64_t)i));
+            auto* id_outp = id_expr->output((int64_t)i)->as<IterDomain>();
+            graph_->initializeVal(id_outp, vg_outp);
+            concrete_ids_.emplace(vg_outp, id_outp);
+          }
+        }
+      }
+      if (uncomputed_producer_groups.empty()) {
+        // All of the defining expressions are proven uncomputable, so mark vg
+        // uncomputable
+        uncomputable_groups.insert(vg);
+        return;
+      } else {
+        // There are some uncomputed producer groups that might be computable,
+        // so try again after processing those producer groups
+        vg_stack.push(vg);
+        for (const ValGroup& next_vg : uncomputed_producer_groups) {
+          vg_stack.push(next_vg);
+        }
+      }
+    };
+
+    while (!vg_stack.empty()) {
+      ValGroup vg = vg_stack.top();
+      vg_stack.pop();
+      if (concrete_ids_.count(vg) != 0) {
+        continue;
+      }
+      propagate(vg);
+    }
+
+    // Now check whether there is a concrete entry for abs_id (i.e. g). If not,
+    // that means this dimension is missing and should not be included in the
+    // loop domain, so return nullptr.
+    auto it = concrete_ids_.find(g);
+    return it == concrete_ids_.end() ? nullptr : it->second;
+  }
+
+ private:
+  const AbstractTensor& abstract_;
```

Comment:
nit: could you change the name to something like `abstract_tensor_`? There are just too many "abstract" something...